lib/propmangle.js

```diff
@@ -100,6 +101,10 @@ function mangle_properties(ast, options) {
 
     var regex = options.regex;
     var ignore_quoted = options.ignore_quoted;
+    var debug = options.debug;
+    
+    // Simulate non-deterministic mangling in debug mode by adding a random number to the name
+    var debug_cache_name = Math.floor(Math.random() * 1000).toString();
```

Comment:
That's a pitfall if you ever refactor to use `let`, since it would scope to just the if block. I could split the `var` and the assignment, but is there much point in that really?